python/lsst/ip/isr/isrFunctions.py

```diff
@@ -244,109 +293,171 @@ def maskITLEdgeBleed(ccdExposure, badAmpDict, itlEdgeBleedSatMinArea=10000,
-
-                # The middle columns of edge bleeds often have
-                # high counts, so  we check there is an edge bleed
-                # by looking at a small image up to 50 pixels from the edge
-                # and around the saturated columns
-                # of the saturated core, and checking its median is
-                # above the sky background by itlEdgeBleedThreshold
-
-                # If the centroid is too close to the edge of the detector
-                # (within 5 pixels), we set the limit to the mean check
-                # to the edge of the detector
-                lowerRangeSmall = int(xCore)-5
-                upperRangeSmall = int(xCore)+5
-                if lowerRangeSmall < 0:
-                    lowerRangeSmall = 0
-                if upperRangeSmall > xmax:
-                    upperRangeSmall = xmax
-                ampImageBG = numpy.median(maskedImage[amp.getBBox()].image.array)
-                edgeMedian = numpy.median(sliceImage[:50, lowerRangeSmall:upperRangeSmall])
-                if edgeMedian > (ampImageBG + itlEdgeBleedThreshold):
-
-                    # We need an estimate of the maximum width
-                    # of the edge bleed for our masking model
-                    # so we now estimate it by measuring the width of
-                    # areas above 60 percent of the saturation level
-                    # close to the edge,
-                    # in a cutout up to 100 pixels from the edge.
-                    subImage = sliceImage[:100, :]
-                    maxWidthEdgeBleed = numpy.max(numpy.sum(subImage > 0.45*satLevel,
-                                                            axis=1))
-
-                    # Mask edge bleed with a
-                    # decaying exponential model
-                    for y in range(200):
-                        edgeBleedHalfWidth = \
-                            int(((maxWidthEdgeBleed)*numpy.exp(-itlEdgeBleedModelConstant*y)
-                                 + widthSat)/2.)
-                        lowerRange = int(xCore)-edgeBleedHalfWidth
-                        upperRange = int(xCore)+edgeBleedHalfWidth
-                        # If the edge bleed model goes outside the detector
-                        # we set the limit for the masking
-                        # to the edge of the detector
-                        if lowerRange < 0:
-                            lowerRange = 0
-                        if upperRange > xmax:
-                            upperRange = xmax
-                        sliceMask[y, lowerRange:upperRange] = satMaskBit
+        widthSat = numpy.sum(fpCore.getSpans().asArray()[int(yCoreFP), :])
+        modelITLEdgeBleed(ccdExposure, xCore,
+                          satLevel, widthSat, itlEdgeBleedThreshold,
+                          itlEdgeBleedModelConstant, saturatedMaskName)
+
+
+def modelITLEdgeBleed(ccdExposure, xCore,
```

Comment:
Is this an internal method?  Underscore?

Reply:
I don't understand the naming or organization now.  This does the masking, not the modeling it looks to me.